tests/components/mqtt/test_init.py

```diff
@@ -420,6 +420,31 @@ async def test_mqtt_publish_action_call_with_template_payload_renders_template(
     mqtt_mock.reset_mock()
 
 
+@pytest.mark.parametrize(
+    ("attr_payload", "payload"),
+    [("b'\\xde\\xad\\xbe\\xef'", b"\xde\xad\xbe\xef"), ("DEADBEEF", "DEADBEEF")],
+)
+async def test_mqtt_publish_action_call_with_raw_data(
+    hass: HomeAssistant,
+    mqtt_mock_entry: MqttMockHAClientGenerator,
+    attr_payload: str,
+    payload: str | bytes,
+) -> None:
+    """Test the mqtt publish action call with a previous rendered template.
+
+    When a template rendered to a binary we try literal_eval to obtain the raw value.
```

Comment:
The docstring is misleading, the test has nothing to do with templates IMHO, it's about special treatment of a payload which is `b"..."`.